### DISCUSSION TITLE
FA of Adiabatic RF in mprage

### DISCUSSION THREAD

**Discussion by zhengliuX** [pulseq/pulseq#44] [General]
Hi, I'm reading writeMPRAGE.m code in seqDemo. 
https://github.com/pulseq/pulseq/blob/master/matlab/demoSeq/writeMPRAGE.m#L39 
```
rf180 = mr.makeAdiabaticPulse('hypsec', sys, 'Duration', 10.24e-3, 'dwell', 1e-5) ;
```
This line tries to create a 180 flip angle adiabatic RF pulseq, I tried to calculate the flip angle of it, and I got:
```
abs(sum(rf180.signal)* (rf180.t(2)-rf180.t(1)))*360
ans =

          287.144751708818

```
Anything I missed here? Or adiabatic RF is a little speicial and not suitable for this equation?

Thanks a lot

**Comment by QingpingChen1201** [comment]
Dear Zheng,
Many thanks for your message! Adiabatic pulses are a class of amplitude- and frequency-modulated RF pulses that are relatively insensitive to B1 inhomogeneity and frequency offset effects. The design of adiabatic pulses is rather complex and utilizes the adiabatic principle. Adiabatic pulses behave differently than non-adiabatic pulses.  For example, the flip angle produced is not simply proportional to B1 magnitude and pulse length, but depends on how the B1 field varies in amplitude and phase throughout the pulse. Likewise, adiabatic pulses cannot be simply scaled or stretched to change their effect; doubling a 90° adiabatic pulse does not produce a 180° inversion pulse. Further reading: https://mriquestions.com/adiabatic-pulses.html#:~:text=Adiabatic%20pulses%20are%20a%20class,the%20B1%20field%20through%20resonance.
Therefore, the flip angle of the adiabatic RF pulse should not be calculated using the formula you mentioned. Pulseq calculates the adiabatic inversion pulse using the sigpy python package. for more details, you can have a look here: https://github.com/mikgroup/sigpy/tree/main.
Best regards,
Qingping

**Comment by zhengliuX** [comment]
Thanks a lot! I will try to dig into the links. 🫰

**Comment by m-a-x-i-m-z** [comment]
This is a generally interesting point to discuss (and actually not an issue), so I'm moving this to the Discussions section on GitHub and we will continue there.

**Comment by m-a-x-i-m-z** [comment]
Dear Zheng, 

As Qingping correctly mentioned, adiabatic pulses are quite different. Normal RF pulses are assumed to create a rotation that is proportional to the amplitude of the B1 field. This proportionality can only be achieved on resonance or as long as the small flip angle approximation (STA) remains valid. Contrary to that adiabatic pulses are designed to achieve a rotation of the spin ensemble, that is largely independent of the B1 field strength. This is a very useful property, but is clearly against the assumption that the flip angle can be proportional to the pulse amplitude. 

So, to summarize the links that Qingping mentions in her post, adiabatic pulses achieve the desired target rotation as long as their RF field is sufficiently strong. The field that is sufficiently strong fulfills the so-called adiabaticity condition, which relates the rate of change of the frequency to the local B1 field amplitude. If you increase the B1 amplitude beyond this point you will always fulfill the condition, but now with a greater margin, e.g. allowing for your coil to have more spatial inhomogeneity. The downside of increasing the pulse amplitude is the increase in the specific absorption rate (SAR). 

In Pulseq we estimate the rate of change of the frequency around the center of the RF pulse (defined as the point with the frequency offset closest to 0) and then rescale the pulse according to the adiabaticity condition. See lines 188-207 in makeAdiabaticPulse.m 

Hope this helps.
--Maxim
